src/main/java/com/fasterxml/jackson/databind/util/NativeImageUtil.java

```diff
@@ -0,0 +1,62 @@
+package com.fasterxml.jackson.databind.util;
+
+import org.graalvm.nativeimage.ImageInfo;
+
+import java.lang.reflect.InvocationTargetException;
+
+/**
+ * Utilities for graal native image support.
+ */
+public class NativeImageUtil {
+    private static final boolean RUNNING_IN_SVM;
+
+    static {
```

Comment:
I think this should be extracted to a separate Class, and let ClassLoader error/exception be used as the signaling mechanism (to avoid dependency to `ImageInfo` from this class). I'll add a separate note on discussion.